facet-json/src/deserialize.rs

```diff
@@ -402,10 +402,60 @@ pub fn from_slice_wip<'input: 'facet, 'facet>(
                                         trace!("Array starting for list ({})!", wip.shape().blue());
                                         reflect!(put_default());
                                     }
+                                    Def::Struct(s) => {
+                                        if s.kind == StructKind::Tuple {
+                                            trace!(
+                                                "Array starting for tuple ({})!",
+                                                wip.shape().blue()
+                                            );
+                                            // Special handling for unit type ()
+                                            if s.fields.is_empty() {
+                                                // Check if the array is empty by peeking at the next token
```

Comment:
This is probably a fine approach, but I think I would have also... Another approach would have been to wait until we get another value, and then when we push it we get a reflect error, and then when we get a reflect error we... I don't know, at that time we generate a nice JSON error instead?